generators/pkg/writer/write.go

```diff
@@ -45,10 +45,48 @@ func WriteToDisk(filename string, writer io.Writer) error {
 	if _, err := writer.Write([]byte("---\n")); err != nil {
 		return err
 	}
-	data, err = yaml.Marshal(&pipeline)
+	data, err = yaml.Marshal(pipeline)
 	if err != nil {
 		return fmt.Errorf("unable to marshal the pipeline %s: %w", pipeline.Name, err)
 	}
+	if _, err := writer.Write(data); err != nil {
+		return err
+	}
+
+	trigger := generator.GenerateTrigger(pipeline)
+	// write TriggerBinding
+	if _, err := writer.Write([]byte("---\n")); err != nil {
+		return err
+	}
+	data, err = yaml.Marshal(trigger.TriggerBinding)
+	if err != nil {
+		return fmt.Errorf("unable to marshal the TriggerBinding %s: %w", trigger.TriggerBinding.Name, err)
+	}
+	if _, err := writer.Write(data); err != nil {
+		return err
+	}
+
+	// write TriggerTemplate
+	if _, err := writer.Write([]byte("---\n")); err != nil {
+		return err
+	}
+	data, err = yaml.Marshal(trigger.TriggerTemplate)
+	if err != nil {
+		return fmt.Errorf("unable to marshal the TriggerTemplate %s: %w", trigger.TriggerTemplate.Name, err)
+	}
+	if _, err := writer.Write(data); err != nil {
+		return err
+	}
+
+	// write EventListener
+	if _, err := writer.Write([]byte("---\n")); err != nil {
```

Comment:
The abundance of writer.Write + yaml.Marshal is starting to look repetitive. Would it make sense to combine this into a func where we just loop over each object?

Reply:
I can break writer.Write + yaml.Marshal out to a separate function. For the objects that need to be written, e.g. Task, Pipeline, TriggerTemplate, ..., should I wrap them in an array like `[]interface{}` or in a struct to loop in the function?

Reply:
sgtm! other option is https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime?tab=doc#Object if you want access to type information.